src/page/index.js

```diff
@@ -19,36 +21,165 @@ import {
   previewImageTitle,
   config,
   cardList,
+  cardTemplate,
+  profileEditImage,
 } from "../utils/constants.js";
 
 const newCardModal = new PopupWithForm("#add-modal", handleCardFormSubmit);
 newCardModal.setEventListeners();
+const confirmationModal = new PopupWithConfirmation("#confirmation-modal");
+confirmationModal.setEventListeners();
 
-const userInformation = new UserInfo(profileTitle, descriptionJob);
-const newProfileModal = new PopupWithForm("#edit-profile-modal", (values) => {
-  userInformation.setUserInfo(values);
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "79fb9209-ee9a-4da9-a138-5a38df01e74c",
+    "Content-Type": "application/json",
+  },
 });
+const profileImage = document.querySelector(".profile__image");
+
+const userInformation = new UserInfo(
+  profileTitle,
+  descriptionJob,
+  profileImage
+);
+const newProfileModal = new PopupWithForm(
+  "#edit-profile-modal",
+  handleProfileFormSubmit
+);
 newProfileModal.setEventListeners();
+
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  // here we change the button text
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      // We need to close only in `then`
+      popupInstance.close();
+    })
+    // we need to catch possible errors
+    // console.error is used to handle errors if you don’t have any other ways for that
+    .catch(console.error)
+    // in `finally` we need to return the initial button text back in any case
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+}
+
+// here is an example of the profile form handling
+function handleProfileFormSubmit(inputValues) {
+  // we create a function that returns a promise
+  function makeRequest() {
+    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
+    return api
+      .editUserInfo(inputValues)
+      .then((userData) => {
+        userInformation.setUserInfo(userData);
+      })
+      .catch(console.error);
```

Comment:
`  .catch(console.error);` is not needed if you use `handleSubmit` (it contains error catching)